### ISSUE TITLE
Changes needed for XTP perfmon counters 

### ISSUE THREAD

**Issue by PiJoCoder** [microsoft/DiagManager#101]
Changes needed for XTP counters - shows version as 2018 instead of 2019 - need to double-check where the issues is - Diagmanager or SQL Server

Also, why do we have 4 XTP counters at the end?
XTP Cursors
XTP Garbage Collection
XTP Phantom processor
XTP Transactions

![image](https://user-images.githubusercontent.com/23519517/120354636-372ee500-c2c8-11eb-999c-892d1e512e13.png)

**Comment by PiJoCoder** [comment]
Based on some research these are actually valid, they exist for the default instance on the system (as long as it supports XTP/in-memory OLTP)

XTP Cursors
XTP Garbage Collection
XTP Phantom processor
XTP Transactions

**Comment by PiJoCoder** [comment]
If you have a named instance sql 2016, the counters would look like this

![image](https://user-images.githubusercontent.com/23519517/178818380-dfbb14e1-b6c2-4edf-b846-a38947609ccc.png)

For a SQL 2017 instance the counters would look like this:

![image](https://user-images.githubusercontent.com/23519517/178818478-43869731-f1d3-4369-a6b9-a529aeec8d74.png)

We still need to double-check the "2018" counters and see if those are valid.

**Comment by PiJoCoder** [comment]
Quick research shows that SQL 2019 is actually using the 2017 in the object name. 

```
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Microsoft SQL Server 2019 (RTM-CU16-GDR) (KB5014353) - 15.0.4236.7 (X64) 
	May 29 2022 15:55:47 
	Copyright (C) 2019 Microsoft Corporation
	Enterprise Edition (64-bit) on Windows 10 Enterprise 10.0 <X64> (Build 19044: ) (Hypervisor)


(1 row affected)

object_name                                                                                                                      counter_name
-------------------------------------------------------------------------------------------------------------------------------- --------------------------------------------------------------------------------------------------------------------------------
SQL Server 2017 XTP Transactions                                                                                                 Transactions created/sec                                                                                                        
SQL Server 2017 XTP Transactions                                                                                                 Read-only transactions prepared/sec                                                                                             
SQL Server 2017 XTP Transactions                                                                                                 Transactions aborted/sec                                                                                                        
SQL Server 2017 XTP Transactions                                                                                                 Transactions aborted by user/sec                                                                                                
SQL Server 2017 XTP Transactions                                                                                                 Transaction validation failures/sec                                                                                             
SQL Server 2017 XTP Transactions                                                                                                 Commit dependencies taken/sec                                                                                                   
SQL Server 2017 XTP Transactions                                                                                                 Cascading aborts/sec                                                                                                            
SQL Server 2017 XTP Transactions                                                                                                 Save points created/sec                                                                                                         
SQL Server 2017 XTP Transactions                                                                                                 Save point rollbacks/sec                                                                                                        
SQL Server 2017 XTP Transactions                                                                                                 Save point refreshes/sec                                                                                                        
SQL Server 2017 XTP Transaction Log                                                                                              Log bytes written/sec                                                                                                           
SQL Server 2017 XTP Transaction Log                                                                                              Log records written/sec                                                                                                         
SQL Server 2017 XTP Cursors                                                                                                      Cursor scans started/sec                                                                                                        
SQL Server 2017 XTP Cursors                                                                                                      Rows returned/sec                                                                                                               
SQL Server 2017 XTP Cursors                                                                                                      Rows touched/sec                                                                                                                
SQL Server 2017 XTP Cursors                                                                                                      Tentatively-deleted rows touched/sec                                                                                            
SQL Server 2017 XTP Cursors                                                                                                      Expired rows touched/sec                                                                                                        
SQL Server 2017 XTP Cursors                                                                                                      Expired rows removed/sec                                                                                                        
SQL Server 2017 XTP Cursors                                                                                                      Dusty corner scan retries/sec (user-issued)                                                                                     
SQL Server 2017 XTP Cursors                                                                                                      Cursor write conflicts/sec                                                                                                      
SQL Server 2017 XTP Cursors                                                                                                      Cursor unique violations/sec                                                                                                    
SQL Server 2017 XTP Cursors                                                                                                      Cursor inserts/sec                                                                                                              
SQL Server 2017 XTP Cursors                                                                                                      Cursor updates/sec                                                                                                              
SQL Server 2017 XTP Cursors                                                                                                      Cursor deletes/sec                                                                                                              
SQL Server 2017 XTP Garbage Collection                                                                                           Rows processed/sec                                                                                                              
SQL Server 2017 XTP Garbage Collection                                                                                           Rows processed/sec (no sweep needed)                                                                                            
SQL Server 2017 XTP Garbage Collection                                                                                           Rows processed/sec (first in bucket)                                                                                            
SQL Server 2017 XTP Garbage Collection                                                                                           Rows processed/sec (first in bucket and removed)                                                                                
SQL Server 2017 XTP Garbage Collection                                                                                           Rows processed/sec (marked for unlink)                                                                                          
SQL Server 2017 XTP Garbage Collection                                                                                           Parallel GC work item/sec                                                                                                       
SQL Server 2017 XTP Garbage Collection                                                                                           Main GC work items/sec                                                                                                          
SQL Server 2017 XTP Garbage Collection                                                                                           Sweep scans started/sec                                                                                                         
SQL Server 2017 XTP Garbage Collection                                                                                           Sweep rows touched/sec                                                                                                          
SQL Server 2017 XTP Garbage Collection                                                                                           Sweep expiring rows touched/sec                                                                                                 
SQL Server 2017 XTP Garbage Collection                                                                                           Sweep expired rows touched/sec                                                                                                  
SQL Server 2017 XTP Garbage Collection                                                                                           Sweep expired rows removed/sec                                                                                                  
SQL Server 2017 XTP Garbage Collection                                                                                           Dusty corner scan retries/sec (GC-issued)                                                                                       
SQL Server 2017 XTP Phantom Processor                                                                                            Phantom scans started/sec                                                                                                       
SQL Server 2017 XTP Phantom Processor                                                                                            Phantom rows touched/sec                                                                                                        
SQL Server 2017 XTP Phantom Processor                                                                                            Phantom expiring rows touched/sec                                                                                               
SQL Server 2017 XTP Phantom Processor                                                                                            Phantom expired rows touched/sec                                                                                                
SQL Server 2017 XTP Phantom Processor                                                                                            Phantom expired rows removed/sec                                                                                                
SQL Server 2017 XTP Phantom Processor                                                                                            Dusty corner scan retries/sec (Phantom-issued)                                                                                  
SQL Server 2017 XTP Storage                                                                                                      Merge Requests Outstanding                                                                                                      
SQL Server 2017 XTP Storage                                                                                                      Merge Policy Evaluations                                                                                                        
SQL Server 2017 XTP Storage                                                                                                      Merges Installed                                                                                                                
SQL Server 2017 XTP Storage                                                                                                      Merges Abandoned                                                                                                                
SQL Server 2017 XTP Storage                                                                                                      Core Merges Completed                                                                                                           
SQL Server 2017 XTP Storage                                                                                                      Total Files Merged                                                                                                              
SQL Server 2017 XTP Storage                                                                                                      Checkpoints Completed                                                                                                           
SQL Server 2017 XTP Storage                                                                                                      Checkpoints Closed                                                                                                              
SQL Server 2017 XTP IO Governor                                                                                                  Log Blocks/sec                                                                                                                  
SQL Server 2017 XTP IO Governor                                                                                                  Io Issued/sec                                                                                                                   
SQL Server 2017 XTP IO Governor                                                                                                  Stale Rate Object Waits/sec                                                                                                     
SQL Server 2017 XTP IO Governor                                                                                                  Insufficient Credits Waits/sec                                                                                                  
SQL Server 2017 XTP IO Governor                                                                                                  Missed Credit Slots                                                                                                             
SQL Server 2017 XTP IO Governor                                                                                                  Total Rate Objects Published                                                                                                    

(57 rows affected)
```

**Comment by PiJoCoder** [comment]
Fixed by #213
